device/src/application/services/impl/DeviceServiceImpl.ts

```diff
@@ -6,49 +6,62 @@ import { Device } from '@/domain/core/Device.js'
 import { DeviceEndpoint } from '@/domain/core/DeviceEndpoint.js'
 import { DeviceFactory } from '@/domain/factories/DeviceFactory.js'
 import { CapabilityType } from '@/domain/core/capabilities/CapabilityType.js'
-import RequestHelper from '@common/utils/RequestHelper.js'
 import { DeviceEventsHub } from '@/application/services/DeviceEventsHub.js'
+import { DeviceEventFactory } from '@common/domain/factories/DeviceEventFactory.js'
+import { Servient } from '@node-wot/core'
+import HttpClientFactory from '@node-wot/binding-http'
 import { CapabilityFactory } from '@/domain/factories/CapabilityFactory.js'
 import { MeasureFactory } from '@common/domain/factories/MeasureFactory.js'
-import { DeviceEventFactory } from '@common/domain/factories/DeviceEventFactory.js'
+import * as console from 'node:console'
+
+const Server = HttpClientFactory.HttpClientFactory
 
 export class DeviceServiceImpl implements DeviceService {
   private readonly repository: DeviceRepository
   private readonly events: DeviceEventsHub
+  private readonly servient: Servient
 
   constructor(repository: DeviceRepository, events: DeviceEventsHub) {
     this.repository = repository
     this.events = events
+    this.servient = new Servient()
+    this.servient.addClientFactory(new Server(null))
   }
 
   async getDeviceCapabilities(deviceId: DeviceId): Promise<DeviceCapability[]> {
     const device: Device = await this.repository.getDeviceById(deviceId)
-    return RequestHelper.get(
-      'http://' + device.endpoint.ipAddress + ':' + device.endpoint.port + '/capabilities'
-    )
-      .then((res: any): DeviceCapability[] => {
-        const capabilities: DeviceCapability[] = []
-        for (let i: number = 0; i < res.data.capabilities.length; i++) {
-          if (res.data.capabilities[i].type == CapabilityType.SENSOR) {
-            capabilities.push(
-              CapabilityFactory.sensoringCapabilityOf(
-                res.data.capabilities[i].capturingInterval,
-                MeasureFactory.createMeasure(
-                  res.data.capabilities[i].measure.type,
-                  res.data.capabilities[i].measure.unit
+    return await this.servient
+      .start()
+      .then(async (WoT: any): Promise<DeviceCapability[]> => {
+        const td = await WoT.requestThingDescription(
+          `http://${device.endpoint.ipAddress}:${device.endpoint.port}/device`
+        )
+        const thing = await WoT.consume(td)
+        return thing.readProperty('status').then(async (data: any): Promise<DeviceCapability[]> => {
```

Comment:
Wonderful

Reply:
Thanks :D

Reply:
Take a look to the [Thing](https://github.com/revue-org/revue-sample-thing)

Reply:
Nice!
I can't see the ```Wot``` object import... I believe is missing

Reply:
these are the imports: one for the Servient and one for the HttpClientServer
import { Servient } from '@node-wot/core'
import HttpClientFactory from '@node-wot/binding-http'
The WoT object is created by the Servient after its start.

If you want you can also take a look to the example in the WoT doc [here](https://github.com/eclipse-thingweb/node-wot#examples). Let me know if i clarified your dubt :D